File: app/src/main/java/art/manguste/android/gamesearch/db/GameDatabase.java
package art.manguste.android.gamesearch.db;

import android.content.Context;
import androidx.room.Database;
import androidx.room.Room;
import androidx.room.RoomDatabase;
import androidx.room.TypeConverter;
import androidx.room.TypeConverters;

@Database(entities = {FavoriteGame.class}, version = 1, exportSchema = false)
@TypeConverters(DateConverter.class)
public abstract class GameDatabase extends RoomDatabase{

    private static final Object LOCK = new Object();
    private static final String DATABASE_NAME = "favoritegames";
    private static volatile GameDatabase sInstance;
    public abstract FavoriteGameDao favoriteGameDao();

    public static synchronized GameDatabase getInstance(Context context) {
        if (sInstance == null) {
            synchronized (LOCK) {
                sInstance = Room
                        .databaseBuilder(context.getApplicationContext(),
                                GameDatabase.class,
                                GameDatabase.DATABASE_NAME)
                        .build();
            }
        }
        return sInstance;
    }
}

File: app/src/main/java/art/manguste/android/gamesearch/db/GameDatabase.kt
package art.manguste.android.gamesearch.db

import android.content.Context
import androidx.room.Database
import androidx.room.Room
import androidx.room.RoomDatabase

@Database(entities = [Game::class], version = 1, exportSchema = false)
abstract class GameDatabase : RoomDatabase(){
    abstract val gameDao: GameDAO

    companion object {
        @Volatile
        private var INSTANCE: GameDatabase? = null

        fun getInstance(context: Context): GameDatabase {
            synchronized(this) {
                var instance = INSTANCE
                if (instance == null) {
                    instance = Room.databaseBuilder(
                            context.applicationContext,
                            GameDatabase::class.java,
                            "Favorite games database")
                            .fallbackToDestructiveMigration()
                            .build()
                    INSTANCE = instance
                }
                return instance
            }
        }
    }
}

